File: poo/Java/RaioPositivo/RaioPositivo.java
import java.util.Scanner;

public class RaioPositivo {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double raio;
        final double PI = 3.1415;
        while (true) {
            System.out.print("Informe o raio: ");
            raio = scanner.nextDouble();
            if (raio > 0) {
                double area = PI * raio * raio;
                System.out.printf("A area eh: %.2f\n", area);
                break;
            }
        }
        scanner.close();
    }
}

File: poo/Cpp/RaioPositivo/RaioPositivo.cpp
#include <iostream>
#include <iomanip>

int main() {
    double raio;
    const double PI = 3.1415;

    // Loop infinito que só será quebrado quando um raio positivo for inserido
    while (true) {
        std::cout << "Informe o raio: ";
        std::cin >> raio;

        if (raio > 0) {
            double area = PI * raio * raio;
            std::cout << "A area eh: " << std::fixed << std::setprecision(2) << area << std::endl;
            break; // Encerra o loop
        }
    }

    return 0;
}